src/shared.ts

```diff
@@ -143,16 +143,27 @@ export const ESGlobalKeys = SetCreate([
     'Intl', // Unstable
 ]);
 
-// These are foundational things that should never be wrapped but are equivalent
+// These are intrinsics that can be reached by syntax, and must be mapped between realms.
 // TODO: revisit this list.
-export const ReflectiveIntrinsicObjectNames = [
-    'Object',
-    'Function',
-    'URIError',
-    'TypeError',
-    'SyntaxError',
-    'ReferenceError',
-    'RangeError',
-    'EvalError',
-    'Error',
-];
+export function extractUndeniableIntrinsics(globalObj: typeof globalThis): any[] {
+    return globalObj.eval(`[
```

Comment:
@erights, do you have a list of all things you consider undeniable intrinsics that we can look at? something other than https://github.com/google/caja/blob/master/src/com/google/caja/ses/repairES5.js which seems outdated.

Additional, I wonder why Errors were not considered undeniable intrinsics since they can be reached by syntax as well via try/catch.

Finally, I think I'm leaning toward the idea that the internal slot issues in membranes are only a deal-breaker when it comes to undeniable intrinsics, if those are using internal slots, then everything is harder. On the other hand, other things (e.g.: Set) are probably ok to continue having identity discontinuity. Unfortunately, Justin proposal is a lot more tricky than we thought, we can chat about it.

Reply:
Attn @jfparadis

Hi @caridy , there is https://github.com/Agoric/SES-shim/blob/master/packages/harden/src/anonIntrinsics.js#L114

However, it does not have `Error.prototype` and `FooError.prototype` for all `FooErrors` that JS itself can throw. It just hadn't occurred to me. I agree --- these are undeniable and should be included on any such list.

@jfparadis, is this code vestigial? It looks like it. AFAICT, it is in fact dead code.

In case it does get removed, the current list is:

```js
    ['Object.prototype', Object.prototype, {}],
    ['Function.prototype', Function.prototype, function foo() {}],
    ['Array.prototype', Array.prototype, []],
    ['RegExp.prototype', RegExp.prototype, /x/],
    ['Boolean.prototype', Boolean.prototype, true],
    ['Number.prototype', Number.prototype, 1],
    ['String.prototype', String.prototype, 'x'],
    ['%Generator%', Generator, aStrictGenerator],
    ['%AsyncGenerator%', AsyncGenerator, aStrictAsyncGenerator],
    ['%AsyncFunction%', AsyncFunctionPrototype, aStrictAsyncFunction],
```

Reply:
Oh. `Promise.prototype` is undeniable since async functions are syntax that directly creates promises.

What else did we miss?

Reply:
@erights that URL is from package `harden`, and is legacy code. Because of numerous issues, it should be deprecated and rethought. The new SES doesn't use it.

Reply:
@jfparadis, ok, that's what I thought. In the new SES, is there any enumeration of undeniables?